lib/Sema/CSFix.cpp

```diff
@@ -170,18 +170,166 @@ MarkExplicitlyEscaping::create(ConstraintSystem &cs, Type lhs, Type rhs,
   return new (cs.getAllocator()) MarkExplicitlyEscaping(cs, lhs, rhs, locator);
 }
 
-bool RelabelArguments::diagnose(const Solution &solution, bool asNote) const {
-  LabelingFailure failure(solution, getLocator(), getLabels());
-  return failure.diagnose(asNote);
+// Find RelabelArgument and MoveOutOfOrder fixes.
+// Result is sorted by priority for emitting aggregated diagnostics.
+static void
+findLabelingFixes(const Solution &solution, Expr *applyAnchor,
+                  SmallVectorImpl<const ConstraintFix *> &foundFixes) {
```

Comment:
Could you please explain why this is necessary?

Reply:
For example, following code

```swift
func f(_ aa: Int, _ bb: Int, cc: Int, dd: Int, ee: Int, ff: Int) {}

f(0, 1, ccx: 2, dd: 3, ff: 5, ee: 4)
```

contains these two errors.

- label wrong: `ccx:` to `cc:`
- out of order label: move `ee:` to before `ff:`

So a `RelabelArgument` object and a `MoveOutOfOrderArgument` object are created at `matchCallArguments`.

And we want to diagnose one aggregated message.

However, both `RelabelArgument::coalesceAndDiagnose` and `MoveOutOfOrderArgument::coalesceAndDiagnose` are invoked by the diagnostic system.

Since `RelabelArgument` and `MoveOutOfOrderArgument` are unaware of each other's existence, they will each output an error message if this continues.

So they need to look for `RelabelArgument` and `MoveOutOfOrderArgument` that are in the same place as theirs.

`findLabelingFixes` is a function for this search.

Reply:
`findLabelingFixes` returns an array of `RelabelArgument` and `MoveOutOfOrderArgument` at the same location.

If there are multiple errors and will output the aggregated error, only one CSFix needs to be output once to prevent do multiple times.

So, the CSFix at first of the array is representative.
Then, the representative CSFix will output the aggregated error.

Following logic in `diagnoseLabelError` is check for this.

```cpp
  if (thisFix != fixes.front())
    return false;
```